### ISSUE TITLE
Allow relative paths for files

### ISSUE THREAD

**Issue by TieSKey** [mcpierce/comixed#452]
Allowing comic files to have relative paths would add a lot of portability to our libraries.

I propose the root of such relative paths be the path to the db file, which in turn should be easily configurable as requested in #451 (so, a "configuration.library.rootFolder" property).

When importing files, just check if the path is currently under the configured "root folder", if it is, make the path relative, if it's not, use the absolute path.

**Comment by mcpierce** [comment]
I'm a little leery of mixing both absolute and relative paths. However, I do like the idea of a relative, rather than absolute, schema for storing comes. Portability of the library would make it easy to share your library among different machines if you're using a stand-alone database, like MySQL, rather than the embedded H2.

**Comment by TieSKey** [comment]
Yeah having both modes sounds weird and is probably prone to errors but the alternative is making everything relative to a single root which might not be appreciated (I wound't mind).
A more robust solution would be to have a hashmap of root folders and every file be relative to one of those. Users can then add or remove such roots. Pretty much what comicrack does now.

To add to the context, the scenario I had in mind when proposing this was the following:
Suppose I have this file structure in my pc:

* comics
  * comiex.db.h2
  * batman
     * files....
  * paturuzu
     * files.....

Later I get a NAS and want to move the library/collection. If everything is relative to the db file I can simply move the entire comics folder and tell comixed the new location. Or even better, if comixed itself is in the comics folder, I can move everything without having to touch any configuration.

It is worth noting this makes it easier to run several instances of the program, pointing to different db files and libraries which might be useful to separate content w/o the need for complex user permission functionality.  Or at least while such functionality is still pending.

**Comment by mcpierce** [comment]
I would definitely prefer the scenario where the database is kept in the root of the library folder.

That said, my question is: how does this handle importing comics? The import feature doesn't require the comics be in a specific location; i.e., you can import from anywhere. So we'll still have a mix of absolute and relative paths.

That said, I don't think it's unrealistic to say "only relative comic files are guaranteed to be 'portable' if you relocate your library".

I know CR supports having comics in multiple locations, but I don't really think of that as a preferable solution. If someone wants to be able to relocate their library, they'll need to adhere to some minimal setup for CX to support that need. And a relative library setup seems a reasonable compromise, n'est pas?

**Comment by TieSKey** [comment]
>  If someone wants to be able to relocate their library, they'll need to adhere to some minimal setup for CX to support that need. And a relative library setup seems a reasonable compromise, n'est pas?

Yeah, sounds like the most sensible solution

Btw, taking a closer look at CR, while it does allow to have multiple "watched folders", files still have absolute paths and file names are either used to id the files o as part of the a hash (bad idea imho) so I couldn't take the chance to clean some file names or I would lose their info (just got an actual NAS) :S

**Comment by mcpierce** [comment]
With the work on the Docker image in #302 I'm more interested in making the library use relative filenames. But we definitely need to think through the possible scenarios and get things right. I'd hate for someone to setup their system and find some flaw that invalidates the whole idea.

**Comment by TieSKey** [comment]
Mmmm, using docker volumes the path to the db file or comics folders can be set from outside even if absolute in the eyes of the program but I wouldn't recommend that approach if the idea is to make it dead easy for non savvy users as docker doesn't make it easy to create volumes in arbitrary locations (it defaults to a hash named folder in some obscure path).

While I do like containers I don't thing it will add much to an already portable program like this one (once the paths are made relative or the consolidation/move function gets implemented).
